tools/update_cli_readme_documentation.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python3
+
+# Tool to automatically update README.md files that contain --help output of c++ TCLAP binaries
+
+import logging
+import argparse
+import os
+import re
+import sys
+from subprocess import Popen, PIPE
+
+if __name__ == '__main__':
+  parser = argparse.ArgumentParser(description='Tool to update README.md CLI documentation.')
+  parser.add_argument(
+      '--binary',
+      help='Relative path to the target binary, for example: "bazel-bin/nighthawk_client".')
+  parser.add_argument(
+      '--readme', help='Relative path to the target documentation file, for example: "README.md"')
+  args = parser.parse_args()
+
+  project_root = os.path.join(os.path.dirname(os.path.join(os.path.realpath(__file__))), "../")
+  # Change directory to avoid TCLAP outputting a full path specification to the binary
+  # in its help output.
+  os.chdir(os.path.realpath(project_root))
+  readme_md_path = os.path.join(project_root, args.readme)
+  process = Popen([args.binary, "--help"], stdout=PIPE)
+  (output, err) = process.communicate()
+  exit_code = process.wait()
+  if exit_code != 0:
+    sys.stderr.writelines(["Failure running --help on with target command"])
+    sys.exit(exit_code)
+
+  # Avoid trailing spaces, as they introduce markdown linting errors.
+  cli_help = [s.strip() for s in output.decode().splitlines()]
+
+  with open(readme_md_path, "r") as f:
+    replaced = re.sub("\nUSAGE\:[^.]*.*%s[^```]*" % args.binary, str.join("\n", cli_help), f.read())
```

Comment:
Can markdown documents inline other docs? That way we could avoid having to do a substitution.

Reply:
I couldn't find a way to accomplish that. We could, however, create links to separate `.md` files, but I'm not sure if that's ideal?

Reply:
The alternative is to make the entire README.md written in the Python `strings.Template` language, load this file and do the substitution with the generate content. WDYT?

Reply:
+1, but as the current approach works and is better then nothing, I created https://github.com/envoyproxy/nighthawk/pull/180#discussion_r340425493 for postponing this to a follow up. I do agree it would be cleaner and less fragile.